Replace the RELEASE.txt file with CHANGELOG.md

loricooperhdf opened the pull request:
An effort to modernize by changing release.txt from a txt file to md. Gemini suggested Changelog instead of sticking with the release.md file name. The filename change is up for debate. For the most part, the work I did here was getting rid of line breaks and placing content into the appropriate sections.

Additional notes to my coworkers: 
- Please see additional comments in this PR where I asked for clarification.
- I need more consistency in when I use backticks to signify something is "code." What constitute code? I think we'll all have to look at our section(s) in this and add in this formatting. Being consistent would look best. 
- This new format has a lot more sections than our old one did. Is your content in the right spot? 
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Replaced `RELEASE.txt` with `CHANGELOG.md`, updating content to markdown format and organizing into detailed sections.
> 
>   - **File Changes**:
>     - Replaced `RELEASE.txt` with `CHANGELOG.md` in markdown format.
>     - Renamed `RELEASE.txt` to `release_archive.txt` for historical data.
>   - **Content Updates**:
>     - Added detailed sections in `CHANGELOG.md` for new features, improvements, bug fixes, breaking changes, and known problems.
>     - Updated content to remove line breaks and organize into appropriate sections.
>   - **Misc**:
>     - Added quick links and contents section in `CHANGELOG.md` for better navigation.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=HDFGroup%2Fhdf5&utm_source=github&utm_medium=referral)<sup> for 98de43af3e9d0efa58feb20932d72808dcbeaee4. You can [customize](https://app.ellipsis.dev/HDFGroup/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

loricooperhdf committed:
Rename RELEASE.txt to release_archive.txt

loricooperhdf committed:
Create CHANGELOG.md

Initial pass of moving release.txt over. Mostly got rid of hard line breaks, added ` for code (TBD on consistency), added tables where appropriate, text changes only when typos/super obvious.

loricooperhdf committed:
Add cmake logo

loricooperhdf committed:
Update CHANGELOG.md

Generally finished with the basic transfer

loricooperhdf committed:
Fixed TOC

loricooperhdf committed:
Fixing TOC links and formatting in Cmake Installations section

loricooperhdf committed:
Split Quick Links and Contents into two lists

loricooperhdf committed:
Additional proofreading, added github issue and PR  links

loricooperhdf commented on release_docs/CHANGELOG.md:
```diff
@@ -0,0 +1,678 @@
+
+   This option enables support for concurrent multithreaded operation of supported API routines. This option also provides threadsafe execution of all other, non-concurrent operations. The 'concurrency' option thus is a superset of the existing 'threadsafe' option. Both options are currently available, although mutually exclusive. As the 'concurrency' code becomes more stable over time, the 'thre ...
+
+   The following API routines support concurrent multithreaded operation:
```
Is it intended that this is left as it is? We might word this differently if no API routines support at release.

lrknox replied:
Some thing like this?

There are no API routines that support concurrent multithreaded operation in this version.

loricooperhdf commented on release_docs/CHANGELOG.md:
```diff
@@ -0,0 +1,678 @@
+
+   This also affects the text output of `H5check_version()` and the libhdf5.settings string stored in the library (for those who use strings(1), etc. to get build info from the binary).
+
+- CMake: Split compiler specific flags into separate files
```
@byrnHDF OK to change this line to 
"Split Cmake compiler specific flags into separate files"

lrknox replied:
@byrnHDF, do we need CMake at all in this line?

lrknox replied:
We can either leave or remove it.

loricooperhdf commented on release_docs/CHANGELOG.md:
```diff
@@ -0,0 +1,678 @@
+
+   The default dataset chunk cache size was increased to 8 MiB (8,388,608 bytes).
+
+- The file format has been updated to 4.0
```
Does this line require additional clarification? It seems like it needs something more to me.

mattjala replied:
I think the following block explaining the change should be sufficient.

loricooperhdf commented on release_docs/CHANGELOG.md:
```diff
@@ -0,0 +1,678 @@
+
+   There is no API compatibility wrapper for this change.
+
+   Fixes GitHub issue [#3506](https://github.com/HDFGroup/hdf5/issues/3506)
```
Should this be in the Bug Fixes section?

loricooperhdf commented on release_docs/CHANGELOG.md:
I think text currently in lines 665-669 is supposed to be here, correct?

lrknox replied:
We should remove the issues with the gif tool and the associated CVE issues after we check on their status.

loricooperhdf commented on release_docs/CHANGELOG.md:
@brtnfld Should Breaking Changes be higher in this document? Maybe before _Bug Fixes_, or even as the first thing before _New Features and Improvements_?

lrknox replied:
Yes

loricooperhdf commented on release_docs/CHANGELOG.md:
@byrnHDF @lrknox Here's where the cdash results would go.

lrknox replied:
Or probably a link to the wiki page where the platforms tested will be shown

loricooperhdf commented on release_docs/CHANGELOG.md:
@brtnfld Keep? Remove?

brtnfld replied:
removed

loricooperhdf commented on release_docs/CHANGELOG.md:
@brtnfld We have _Fortran API_, _Fortran High Level APIs_, and _F90 APIs_ as three (empty) sections?

brtnfld replied:
Correct

loricooperhdf commented on release_docs/CHANGELOG.md:
These lines (665-669) seemingly belong around line 413 instead of here.

lrknox committed:
Update CMakeInstallation.cmake to install CMAKELOG.md instead of
RELEASE.txt.

lrknox committed:
Remove duplicate line 110 in release_docs/CHANGELOG.md

Co-authored-by: ellipsis-dev[bot] <65095814+ellipsis-dev[bot]@users.noreply.github.com>

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into Changelog_instead_of_release

lrknox committed:
Merged changes from develop.
Added */CHANGELOG.md \ to list in doxygen/Doxyfile.in of file patterns to be excluded from
doxygen processing.

byrnHDF commented:
I think there are 36 more instances that need changed:
.github/PULL_REQUEST_TEMPLATE/pull_request_template.md
release_docs/NEWSLETTER_README.txt (2 matches)
release_docs/README.md (13 matches)
release_docs/RELEASE_PROCESS.md (15 matches)
CONTRIBUTING.md (2 matches)
README.md (2 matches)

brtnfld committed:
Update CHANGELOG.md

Format clean-up

brtnfld commented on release_docs/CHANGELOG.md:
@byrnHDF @lrknox why is this section needed the release.txt

byrnHDF replied:
We could put this section in AutotoolsToCMakeOptions.md and leave a single line:
See the AutotoolsToCMakeOptions.md file for CMake options of former Autotools options.

byrnHDF commented on release_docs/CHANGELOG.md:
DEFAULT_CDELEMTS is the correct name of the define

byrnHDF commented on release_docs/CHANGELOG.md:
I think this section should be first

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into Changelog_instead_of_release

lrknox committed:
Added RELEASE.txt entries through 9/11/2025.

lrknox committed:
Merge branch 'Changelog_instead_of_release' of https://github.com/loricooperhdf/hdf5 into Changelog_instead_of_release

lrknox committed:
Replace 30+ RELEASE.txt to CHANGELOG.md references in six documentation
files.

lrknox committed:
Moved CMake Installations section from the bottom of CHANGELOG.md to the
top of AutotoolsToCMakeOptions.md with a reference in CHANGELOG.md's
Breaking Changes section.
Removed the list of 3 CVEs related to the hl gif tools as those tools
have been removed.

lrknox committed:
Add changes from RELEASE.txt through develop commit 957a5091, 9/16/2025.

lrknox committed:
Merge branch 'develop' of https://github.com/HDFGroup/hdf5 into Changelog_instead_of_release
